crates/optimism/bin/src/main.rs

```diff
@@ -60,6 +65,33 @@ struct Args {
     pub proofs_history_window: u64,
 }
 
+async fn launch_node_with_storage<S>(
```

Comment:
With this particular design we need to initialize Proof Storage regardless of proof history is enabled or not. Because we are checking the `proofs_history_enabled` inside this function.

Reply:
can you pls open a debt issue for improving this design permalinking to this code @sadiq1971